### PR TITLE
Expand on bare-metal docs relating to infrastructure requirements

### PR DESCRIPTION
**Pull request by zacharyblasczyk**
## Description

What does the pull request do? If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Ticket

Does this PR fix an existing issue? If yes, provide a link to the ticket here:

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [x] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [x] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [x] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/hosting/hosting-options/self-managed.md [hunk 1]
```diff
@@ -28,3 +28,79 @@ This is a fully self-hosted solution where W&B server is running in a scalable d
 - a redis cache (optional)
 
 W&B can provide recommendations for compatible database engines, object stores and have an experienced team to help with the installation process. The complexity of administrating a database, creating and maintaining a distributed object storage system adds additional overhead to the customer's IT/DevOps/MLOps teams. When possible, W&B recommends using W&B managed cloud solutions for better user experience.
+
+### Infrastructure Guidelines
+
+It is highly recommended to deploy W&B into a kubernetes cluster to fully take advantage of all the features of the product and to utilize the standardized [helm interface](https://github.com/wandb/helm-charts) that we provide.
+
+If kubernetes is unavailable, W&B can be installed onto a bare-metal server and configured manually, however, future features may be broken out into k8s native or custom resource definitions and may not be backported into the standalone Docker Container.
```
Comment (zacharyblasczyk) [inline docs/guides/hosting/hosting-options/self-managed.md]: future features may be broken out into k8s native or custom resource definitions and may not be backported into the standalone Docker Container.  

I was thinking about Prometheus as an example.

I don't think there has been a shift to a docker composed based deployment.

File: docs/guides/hosting/hosting-options/self-managed.md [hunk 2]
```diff
@@ -28,3 +28,80 @@ This is a fully self-hosted solution where W&B server is running in a scalable d
...
+| Memory (GiB)               | 8                                 |
+
+This ensures that W&B has sufficient disk space to process W&B server application data and store temporary logs before they are externalized. It also ensures fast and reliable data transfer, the necessary processing power and memory for smooth operation, and that W&B will not be affected by any noisy neighbors. 
+
+It is important to keep in mind that these specifications are minimum requirements, and actual resource needs may vary depending on the specific usage and workload of the W&B application. Monitoring the resource usage and performance of the application is critical to ensure that it operates optimally and to make adjustments as necessary.
+
+
+### Database Server
+
+W&B recommends a [MySQL 8](../how-to-guides/bare-metal.md#mysql-80) database as a metadata store of all project, report, artifact, and run data. The shape of the ML practitioners parameters and metadata will greatly affect the performance of the database. The database is typically incrementally written to as SMLE’s track their training runs and is more read heavy when queries are executed in reports and dashboard.
```
Comment (venky-wandb) [inline docs/guides/hosting/hosting-options/self-managed.md]: - Just saying `database as a metadata store.` would probably suffice here. Can remove this part `of all project, report, artifact, and run data`
- SMLE is an internal name we have for our success MLEs. We should change this to practitioners in all places for consistency.

**Comment by venky-wandb** [comment]
@wandb-zacharyblasczyk This entire page should be under `how-to guides > bare-metal` section. The current location is just an intro page for what bare-metal. Actual configuration details should be part of the how-to guide.

File: docs/guides/hosting/hosting-options/self-managed.md [hunk 3]
```diff
@@ -28,3 +28,80 @@ This is a fully self-hosted solution where W&B server is running in a scalable d
...
+| Memory (GiB)               | 32                                |
+
+
+Again, we recommend monitoring the resource usage and performance of the database to ensure that it operates optimally and to make adjustments as necessary.
+
+Additionally, we recommend the following [parameter overrides](../how-to-guides/bare-metal.md#mysql-80) to tune the DB for MySQL 8.
+
+### Object Storage
+
+W&B has no hardware requirements for the on premises storage backend as long as it supports an S3 API interface, Signed URL’s, and a CORS policy. We recommend specing the storage array to the current needs of your ML practitioners and to capacity plan on a regular cadence.
```
Comment (venky-wandb) [inline docs/guides/hosting/hosting-options/self-managed.md]: I think we could rephrase this to say, `W&B is compatible with an object storage that supports S3 API interface, Signed URLs and CORS.` 

No hardware requirements part might not necessarily be true in some cases so just tweaking it to say what we need.

**Comment by zacharyblasczyk** [comment]
@venky-wandb, I kind of agree and disagree.

I started with that page but found that it really doesn't fit. 

This page gives a high level narrative of _what_ is needed for bare metal and brief details about server requirements and object storage options/reqs, the other page gives a detailed explanation of _how_ to configure those pieces for bare metal regardless of what they are.

Let me know what you think.

**Comment by zacharyblasczyk** [comment]
@venky-wandb, I moved the CORS policy out as it was more of a detailed config item.

File: docs/guides/hosting/hosting-options/self-managed.md [hunk 4]
```diff
@@ -28,3 +28,65 @@ This is a fully self-hosted solution where W&B server is running in a scalable d
 - a redis cache (optional)
 
 W&B can provide recommendations for compatible database engines, object stores and have an experienced team to help with the installation process. The complexity of administrating a database, creating and maintaining a distributed object storage system adds additional overhead to the customer's IT/DevOps/MLOps teams. When possible, W&B recommends using W&B managed cloud solutions for better user experience.
+
+### Infrastructure Guidelines
+
+It is highly recommended to deploy W&B into a kubernetes cluster to fully take advantage of all the features of the product and to utilize the standardized [helm interface](https://github.com/wandb/helm-charts) that we provide.
```
Comment (ngrayluna) [inline docs/guides/hosting/hosting-options/self-managed.md]: QQ: would customers otherwise not have access to all W&B features and/or the helm interface?

File: docs/guides/hosting/hosting-options/self-managed.md (hunk 1 above)
Comment (ngrayluna) [reply docs/guides/hosting/hosting-options/self-managed.md]: How certain are we that future features might not be backported into the standalone Docker container?

If fairly positive, maybe we can say something like "W&B features will no longer be backported into standalone Docker Container"... (or something like that).

**Comment by venky-wandb** [comment]
> @venky-wandb, I kind of agree and disagree.
> 
> I started with that page but found that it really doesn't fit.
> 
> This page gives a high level narrative of _what_ is needed for bare metal and brief details about server requirements and object storage options/reqs, the other page gives a detailed explanation of _how_ to configure those pieces for bare metal regardless of what they are.
> 
> Let me know what you think.

I would like to keep the Hosting Options view clean and just have a 30,000 foot view of the deployment type. Specifics of the deployment type should go into the configuration section. `Infrastructure Guidelines` sounds like it'd be more closer to the Guides section rather than the hosting options section. Also, hosting options is the page we expect a lot of user personas to visit including decision makers, ml engineers etc. - that's one of the main reasons to keep it generic and not go into too much depth and overwhelm with lots of info. Guides is a section which we expect MLOps Admins would visit who'd be able to consume the infra info meaningfully. Hope that makes sense! :)

File: docs/guides/hosting/hosting-options/self-managed.md (hunk 1 above)
Comment (venky-wandb) [reply docs/guides/hosting/hosting-options/self-managed.md]: To avoid any confusion, I'd suggest to make the following changes:
We will split this into two sentences. 

First,
```
If kubernetes is unavailable, W&B can be installed onto a bare-metal server and configured manually.
```

Second,
```
Note that, future W&B features may require leveraging k8s native capabilities and hence would require a K8s first environment.
```

**Comment by ngrayluna** [comment]
Great response @venky-wandb and @wandb-zacharyblasczyk. Funny enough,  I agree with both of ya'll. 

It makes sense to keep the hosting options clean. 
But it also makes sense that the infra guidelines don't really belong in the *how* portion of our docs.

It feels like these infra guidelines should be their own thing page? Do we have infra guidelines for on-prem private cloud as well?

**Review by ngrayluna — APPROVED** [review]

**Review by venky-wandb — APPROVED** [review]
Looks solid, thank you Zach!! 🔥
